### PR TITLE
spice-vdagentd service : init at version 0.16.0 

### PR DESCRIPTION
**Pull request by aboseley**
a service for qemu guests

### CODE REVIEW

File: nixos/modules/services/misc/spice-vdagentd.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+let
+  cfg = config.services.spice-vdagentd;
+in
+{
+  options = {
+    services.spice-vdagentd = {
+      enable = mkOption {
+        default = false;
+        description = ''
+          Whether to enable spice guest vdagent  daemon.
```
Comment (pSub) [inline nixos/modules/services/misc/spice-vdagentd.nix]: Nitpick: There is one space too much.

File: pkgs/applications/virtualization/spice-vdagent/default.nix [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  src = fetchurl {
+    url = http://www.spice-space.org/download/releases/spice-vdagent-0.16.0.tar.bz2;
+    sha256 = "0z8gwc5va2i64mjippavmxajdb9az83ffqyhlbynm6dxw131d5av";
+  };
+  postPatch = ''
+    substituteInPlace data/spice-vdagent.desktop --replace /usr $out
+  '';
+  buildInputs = [ pkgconfig alsaLib spice_protocol glib
+                  libpciaccess libxcb libXrandr libXinerama libXfixes ] ;
+}
```
Comment (pSub) [inline pkgs/applications/virtualization/spice-vdagent/default.nix]: Please add meta data.

File: pkgs/applications/virtualization/spice-vdagent/default.nix [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{stdenv, fetchurl, pkgconfig, alsaLib, spice_protocol, glib,
+ libpciaccess, libxcb, libXrandr, libXinerama, libXfixes}:
+stdenv.mkDerivation {
+  name = "spice-vdagent-0.16.0";
+  src = fetchurl {
+    url = http://www.spice-space.org/download/releases/spice-vdagent-0.16.0.tar.bz2;
```
Comment (pSub) [inline pkgs/applications/virtualization/spice-vdagent/default.nix]: You can use string interpolation here to avoid repeating the name.

**Comment by aboseley** [comment]
Hi Pascal,
I've incorporated your comments 
~Adam

File: nixos/modules/services/misc/spice-vdagentd.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  cfg = config.services.spice-vdagentd;
+in
+{
+  options = {
+    services.spice-vdagentd = {
+      enable = mkOption {
```
Comment (jagajaga) [inline nixos/modules/services/misc/spice-vdagentd.nix]: Use `mkEnableOption`.

File: pkgs/applications/virtualization/spice-vdagent/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    url = "http://www.spice-space.org/download/releases/${name}.tar.bz2";
+    sha256 = "0z8gwc5va2i64mjippavmxajdb9az83ffqyhlbynm6dxw131d5av";
+  };
+  postPatch = ''
+    substituteInPlace data/spice-vdagent.desktop --replace /usr $out
+  '';
+  buildInputs = [ pkgconfig alsaLib spice_protocol glib
+                  libpciaccess libxcb libXrandr libXinerama libXfixes ] ;
+  meta = {
+    description = "enhanced SPICE integration for linux QEMU guest";
```
Comment (jagajaga) [inline pkgs/applications/virtualization/spice-vdagent/default.nix]: Please start with a capital letter.

File: nixos/modules/services/misc/spice-vdagentd.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  cfg = config.services.spice-vdagentd;
+in
+{
+  options = {
+    services.spice-vdagentd = {
+      enable = mkEnableOption "Whether to enable spice guest vdagent daemon";
```
Comment (jagajaga) [inline nixos/modules/services/misc/spice-vdagentd.nix]: Just write `spice guest vdagent daemon` without `Whether to enable` :)

**Comment by Profpatsch** [comment]
@aboseley Can you please rebase it on the current master?

**Comment by aboseley** [comment]
@Profpatsch  -  Done

**Pull request merged**